source/common/stats/stats_impl.cc

```diff
@@ -10,6 +10,42 @@
 namespace Envoy {
 namespace Stats {
 
+namespace {
+
+// Round val up to the next multiple of the natural alignment.
+// Note: this implementation only works because 8 is a power of 2.
+size_t roundUpMultipleNaturalAlignment(size_t val) {
+  const size_t multiple = alignof(RawStatData);
+  return (val + multiple - 1) & ~(multiple - 1);
+}
+
+} // namespace
+
+size_t RawStatData::size() {
+  return roundUpMultipleNaturalAlignment(sizeof(RawStatData) + nameSize());
+}
+
+size_t& RawStatData::getMaxNameLength(size_t configured_size) {
+  // Like CONSTRUCT_ON_FIRST_USE, but non-const so that the value can be changed by tests
+  static size_t size = configured_size;
+  return size;
+}
+
+void RawStatData::configure(Server::Options& options) {
+  const size_t configured = options.maxStatNameLength();
+  RELEASE_ASSERT(configured > 0);
```

Comment:
Can we gracefully fail this in options parsing that this is maybe >= the current default? Otherwise I think usability will be very bad.

Reply:
done